qcodes_contrib_drivers/drivers/Attocube/AttocubeController.py

```diff
@@ -0,0 +1,324 @@
+                            snapshot_get=False
+                            )
+        for axis, idx in self.axes.items():
+            self.voltage_limits.update({axis: self.metadata['voltage_limits'][temp][axis]}) # Volts
+            # axis mode
+            self.add_parameter('mode_ax{}'.format(idx),
+                                label='{} axis mode'.format(axis),
+                                unit='',
+                                get_cmd='getm {}'.format(idx),
+                                set_cmd='setm {} {{}}'.format(idx),
+                                vals=vals.Enum('gnd', 'inp', 'cap', 'stp', 'off', 'stp+', 'stp-'),
+                                get_parser=self._mode_parser,
+                                snapshot_get=False
+                                )
+            # axis voltage, with limits set according to temperature mode
+            self.add_parameter('voltage_ax{}'.format(idx),
+                                label='{} axis voltage'.format(axis),
+                                unit='V',
+                                get_cmd='getv {}'.format(idx),
+                                set_cmd='setv {} {{:.3f}}'.format(idx),
+                                vals=vals.Numbers(min_value=0, max_value=self.voltage_limits[axis]),
+                                get_parser=self._voltage_parser,
+                                snapshot_get=True
+                                )
+            # axis frequency
+            self.add_parameter('freq_ax{}'.format(idx),
+                                label='{} axis frequency'.format(axis),
+                                unit='Hz',
+                                get_cmd=(lambda idx=idx: self._get_freq(idx)),
+                                set_cmd='setf {} {{:.0f}}'.format(idx),
+                                vals=vals.Numbers(min_value=1, max_value=10000),
+                                get_parser=self._freq_parser,
+                                snapshot_get=True
+                                )
+            # axis capacitance (gettable, will fail if Attocubes are grounded with e.g. a GND cap)
+            self.add_parameter('cap_ax{}'.format(idx),
+                               label='{} axis capacitance'.format(axis),
+                               unit='nF',
+                               get_cmd=(lambda idx=idx: self._get_cap(idx)),
+                               get_parser=self._cap_parser,
+                               snapshot_get=False
+                              )
+    
+    def ask_raw(self, cmd: str) -> str:
+        """Query instrument with cmd and return response.
+        
+        Args:
+            cmd: Command to write to controller.
+
+        Returns:
+            (str): Response of Attocube controller to the query `cmd`.
+        """
+        self.device_clear()
+        response = super().ask_raw(cmd)
+        status = super().ask_raw(cmd)
+        self.check_response(status)
+        time.sleep(0.2)
```

Comment:
Why do we need to sleep?

Reply:
You're right that we might not need to sleep. As noted in the docstring, I had a hard time getting serial communication to work reliably with these instruments when I originally wrote this driver. I know that it works reliably with the sleep. I no longer have access to the instrument itself, so I unfortunately can't test the driver without the sleep.

Reply:
Ok, no problem it can stay. I just wanted to know if there is a specific reason.